### PR TITLE
bump time library to 1.9.2

### PR DESCRIPTION
**Pull request by molt3nrock**
No CI build.  but it compiles and tested locally on my computer.

I have to update base, network, hs-socks, hs-connection library to use latest time library.

### CODE REVIEW

**Comment by molt3nrock** [comment]
I found out that we don't have to update any other library except time.

so below cabal.project is enough for upgrading it.

```
packages: .

allow-newer:  time 

constraints: time >= 1.9.2
```

if you want minimize dependency upgrade, I'll fix cabal.project accordingly.
let me know.

**Comment by molt3nrock** [comment]
stack configuration has been modified to pull the time package from https://github.com/haskell/time

Below warnings from stack are about relaxing constraints.
> WARNING: Ignoring out of range dependency (allow-newer enabled): time-1.9.2. unix requires: >=1.2 && <1.9
> WARNING: Ignoring out of range dependency (allow-newer enabled): time-1.9.2. directory requires: >=1.4 && <1.9
> WARNING: Ignoring out of range dependency (allow-newer enabled): time-1.9.2. Cabal requires: >=1.4 && <1.9
> WARNING: Ignoring out of range dependency (allow-newer enabled): time-1.9.2. http-api-data requires: >=1.4.2 && <1.9
> WARNING: Ignoring out of range dependency (allow-newer enabled): time-1.9.2. req requires: >=1.2 && <1.9
> WARNING: Ignoring out of range dependency (allow-newer enabled): time-1.9.2. discord-haskell requires: >=1.8.0.2 && <1.9

File: stack.yaml [hunk 1]
```diff
@@ -43,6 +43,12 @@ extra-deps:
 - hookup-0.2.2
 - irc-core-2.5.0
 - discord-haskell-0.7.1
+# use time-1.9.2 from github
+- github: haskell/time
+  commit: 9e96c26132fef01a3113c8b152b1be96c0eccd86
```
Comment (rexim) [inline stack.yaml]: Why can't we just do `time-1.9.2` like everything above?

Comment (molt3nrock) [reply stack.yaml]: time-1.9.2 doesn't exists in the stackage snapshot, so we must pull them other than stackage.

Comment (molt3nrock) [reply stack.yaml]: Nevermind.  No need pull from github. 😅

**Comment by rexim** [comment]
@molt3nrock is that `cabal.project` file a part of the Nix-style builds? I see it being mentioned here: https://cabal.readthedocs.io/en/latest/nix-local-build.html

We have a [Dockerfile](https://github.com/tsoding/HyperNerd/blob/cb47dc4a677dd0b23ab9b57a4efe8e3ae6eda2b2/Dockerfile) that uses old style builds. Do we have to update it to Nix-style to make use of `cabal.project`? (updating Dockerfile is outside of the scope of this PR, of course, just asking)

**Comment by rexim** [comment]
https://github.com/tsoding/HyperNerd/pull/607#issuecomment-493673990 was not addressed within two weeks. According to [Contribution Rules](https://github.com/tsoding/HyperNerd/blob/fceec5df32b77adc68a791d25d2cf72c2c582ae0/CONTRIBUTING.md#pull-requests) this PR is dimmed neglected. Feel free try again at any time.

**Pull request closed without merging**